src/cache.rs

```diff
@@ -145,19 +149,26 @@ impl BlockchainDbMeta {
     /// Sets the [BlockEnv] of this instance using the provided [alloy_rpc_types::Block]
     pub fn with_block<T: TransactionResponse, H: BlockHeader>(
         mut self,
+        chain: Chain,
         block: &alloy_rpc_types::Block<T, H>,
     ) -> Self {
+        let blob_base_fee_update_fraction =
+            match EthereumHardfork::from_chain_and_timestamp(chain, block.header.timestamp()) {
+                Some(EthereumHardfork::Cancun) => BLOB_BASE_FEE_UPDATE_FRACTION_CANCUN,
+                _ => BLOB_BASE_FEE_UPDATE_FRACTION_PRAGUE,
+            };
+
         self.block_env = BlockEnv {
-            number: block.header.number(),
+            number: U256::from(block.header.number()),
             beneficiary: block.header.beneficiary(),
-            timestamp: block.header.timestamp(),
+            timestamp: U256::from(block.header.timestamp()),
             difficulty: U256::from(block.header.difficulty()),
             basefee: block.header.base_fee_per_gas().unwrap_or_default(),
             gas_limit: block.header.gas_limit(),
             prevrandao: block.header.mix_hash(),
             blob_excess_gas_and_price: Some(BlobExcessGasAndPrice::new(
                 block.header.excess_blob_gas().unwrap_or_default(),
-                false,
+                blob_base_fee_update_fraction,
```

Comment:
do we really need this, this previously also just defaulted to pre-prague and I believe we only use this to tag the metadata?
should we just use BLOB_BASE_FEE_UPDATE_FRACTION_PRAGUE and remove the chainid arg?

Reply:
In the future w/ Osaka there will be another `BLOB_BASE_FEE_UPDATE_FRACTION_*` update hence why I did add it for completeness / correctness

Reply:
Ended up removing it in https://github.com/foundry-rs/foundry-fork-db/pull/53/commits/e819e0a5eb48254d6546e51eaad8edb68946bcb3, restored test in https://github.com/foundry-rs/foundry-fork-db/pull/53/commits/2212816166bf0314f2190ad26414da1019967600 to show equivalence for end users